File: mobile/roadsos/lib/features/incidents/domain/models/incident.dart
class Incident {
  const Incident({
    required this.id,
    required this.timestamp,
    required this.latitude,
    required this.longitude,
    required this.alertSent,
  });

  final String id;
  final DateTime timestamp;
  final double latitude;
  final double longitude;
  final bool alertSent;

  Incident copyWith({
    String? id,
    DateTime? timestamp,
    double? latitude,
    double? longitude,
    bool? alertSent,
  }) {
    return Incident(
      id: id ?? this.id,
      timestamp: timestamp ?? this.timestamp,
      latitude: latitude ?? this.latitude,
      longitude: longitude ?? this.longitude,
      alertSent: alertSent ?? this.alertSent,
    );
  }
}

File: backend/app/schemas/incident.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel

from app.models.incident import IncidentStatus


class IncidentOut(BaseModel):
    id: UUID
    latitude: float
    longitude: float
    address_text: str | None
    status: IncidentStatus
    triggered_at: datetime

    model_config = {"from_attributes": True}
